File: databinding-base/src/main/java/com/hangox/databinding/context/LayoutProvider.java
package com.hangox.databinding.context;

import android.support.annotation.LayoutRes;

/**
 * Created With Android Studio
 * User hangox
 * Email liang.hanguang93@gmail.com
 * Date 2017/11/24
 * Time 下午10:56
 * 可以提供Layout
 */

public interface LayoutProvider {


    @LayoutRes  int provideLayoutId();
}

File: databinding-base/src/main/java/com/hangox/starter/context/LayoutProvider.kt
package com.hangox.starter.context


import androidx.annotation.LayoutRes

/**
 * Created With Android Studio
 * User hangox
 * Email liang.hanguang93@gmail.com
 * Date 2017/11/24
 * Time 下午10:56
 * 可以提供Layout
 */

interface LayoutProvider {


    @LayoutRes
    fun provideLayoutId(): Int
}
